Purchases/Networking/Backend.swift

```diff
@@ -605,11 +634,14 @@ private extension Backend {
             let finishable = statusCode < HTTPStatusCodes.internalServerError.rawValue
             var extraUserInfo = [ErrorDetails.finishableKey: finishable] as [String: Any]
             extraUserInfo.merge(subscriberAttributesErrorInfo) { _, new in new }
-            let backendErrorCode = BackendErrorCode(maybeCode: response?["code"])
-            let message = response?["message"] as? String
-            let responseError = ErrorUtils.backendError(withBackendCode: backendErrorCode,
+            let backendErrorCode = BackendErrorCode(maybeCode: maybeResponse?["code"])
+            let message = maybeResponse?["message"] as? String
+            var responseError = ErrorUtils.backendError(withBackendCode: backendErrorCode,
                                                         backendMessage: message,
                                                         extraUserInfo: extraUserInfo as [NSError.UserInfoKey: Any])
+            responseError = ErrorUtils.attach(subError: maybeError,
```

Comment:
starting to feel like we could use some kind of builder for errors

Reply:
Yeah... although I'm averse to the builder pattern for some reason. I feel like it's not swifty, but I can't think of an alternative at the moment.